src/main/java/org/opensearch/knn/index/codec/KNN990Codec/NativeEngines990KnnVectorsWriter.java

```diff
@@ -261,12 +262,27 @@ private <T, C> void trainAndIndex(
 
         StopWatch stopWatch = new StopWatch();
         stopWatch.start();
-        indexOperation.buildAndWrite(writer, knnVectorValues);
+        indexOperation.buildAndWrite(writer, knnVectorValues, totalLiveDocs);
         long time_in_millis = stopWatch.totalTime().millis();
         graphBuildTime.incrementBy(time_in_millis);
         log.warn("Graph build took " + time_in_millis + " ms for " + operationName);
     }
 
+    /**
+     * The {@link KNNVectorValues} will be exhausted after this function run. So make sure that you are not sending the
+     * vectorsValues object which you plan to use later
+     */
+    private int getLiveDocs(KNNVectorValues<?> vectorValues) throws IOException {
```

Comment:
This seems out of place, ideally we should be tackling this when we create `KNNVectorValues` after the merge. Having this along with `totalLiveDocs()` function in KNNVectorValues creates ambiguity. What's the blocker in making sure we don't need this as a separate parameter and be correctly encapsulated in `KNNVectorValues`?

Reply:
The only way to get this via the totalLiveDocs is we figure out a way to clone the iterators. But that I am keeping for later, as I need to check on each and every iterator around how to clone it. Hence I came up with this.